src/unaudited/examples/ECDSAStakeRegistryPermissioned.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import {ISignatureUtils} from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import {ECDSAStakeRegistry} from "../ECDSAStakeRegistry.sol";
+import {IDelegationManager} from "eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+
+/// @title ECDSA Stake Registry with an Operator Allowlist
+/// @notice This contract extends ECDSAStakeRegistry by adding functionality to allowlist and remove operators
+contract ECDSAStakeRegistryPermissioned is ECDSAStakeRegistry {
+    mapping(address => bool) public allowlistedOperators;
+
+    /// @dev Custom error to signal that an operator is not allowlisted.
+    error OperatorNotAllowlisted();
+
+    constructor(IDelegationManager _delegationManager) ECDSAStakeRegistry(_delegationManager) {
+        // _disableInitializers();
+    }
+
+    /// @notice Adds an operator to the allowlisted operator set
+    /// @dev An allowlisted operator isn't a part of the operator set. They must subsequently register themselves
+    /// @param _operator The address of the operator to be allowlisted
+    function permitOperator(address _operator) external onlyOwner {
+        allowlistedOperators[_operator] = true;
+    }
+
+    /// @notice Revokes an operator's permission and deregisters them
+    /// @param _operator The address of the operator to remove from the allowlist and deregistered.
+    function revokeOperator(address _operator) external onlyOwner {
+        delete allowlistedOperators[_operator];
+        _deregisterOperator(_operator);
```

Comment:
seems like this line would end up reverting if the `_operator` is not actively registered.
not a major issue but would recommend addressing this

Reply:
yeah nice callout, updated https://github.com/Layr-Labs/eigenlayer-middleware/pull/194/commits/a72f7e910be201ce010c23387d817ef9053b1253